plugins/modules/cloud/misc/proxmox_storage_info.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: Tristan Le Guern (@Aversiste) <tleguern at bouledef.eu>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: proxmox_storage_info
+short_description: Retrieve information about one or more Proxmox VE storages
+version_added: 2.2.0
+description:
+  - Retrieve information about one or more Proxmox VE storages.
+options:
+  storage:
+    description:
+      - Only return informations on a specific storage
+    aliases: ['name']
+    type: str
+  type:
+    description:
+      - Filter on a specifc storage type
+    type: str
+author: Tristan Le Guern (@Aversiste)
+extends_documentation_fragment: community.general.proxmox.documentation
+notes:
+  - Storage specific options can be returned by this module, please look at the documentation https://pve.proxmox.com/wiki/Storage.
+'''
+
+
+EXAMPLES = '''
+- name: List existing storages
+  community.general.proxmox_storage_info:
+    api_host: helldorado
+    api_user: root@pam
+    api_password: "{{ password | default(omit) }}"
+    api_token_id: "{{ token_id | default(omit) }}"
+    api_token_secret: "{{ token_secret | default(omit) }}"
+  register: proxmox_storages
+
+- name: List NFS storages only
+  community.general.proxmox_storage_info:
+    api_host: helldorado
+    api_user: root@pam
+    api_password: "{{ password | default(omit) }}"
+    api_token_id: "{{ token_id | default(omit) }}"
+    api_token_secret: "{{ token_secret | default(omit) }}"
+    type: nfs
+  register: proxmox_storages_nfs
+
+- name: Retrieve information about the lvm2 storage
+  community.general.proxmox_storage_info:
+    api_host: helldorado
+    api_user: root@pam
+    api_password: "{{ password | default(omit) }}"
+    api_token_id: "{{ token_id | default(omit) }}"
+    api_token_secret: "{{ token_secret | default(omit) }}"
+    storage: lvm2
+  register: proxmox_storage_lvm
+'''
+
+
+RETURN = '''
+proxmox_storage:
+  description: List of storage pools.
+  returned: always
```

Comment:
```suggestion
  returned: on success
```
You have at least one `module.fail_json` which does not return this :)

Reply:
You are right, this was too quickly copy/pasted 🤦